helix-core/src/test/java/org/apache/helix/common/ZkTestBase.java

```diff
@@ -719,6 +722,20 @@ public void cleanupLiveInstanceOwners() {
       clientMap.clear();
     }
     _liveInstanceOwners.clear();
+
+    // wait 2 seconds for threads to be teared down.
+    // this reduce false alert.
+    Thread.sleep(1000);
+    boolean status = false;
+    try {
+       status = ThreadLeakageChecker.afterClassCheck(testClassName);
+    } catch (Exception e) {
+      System.out.println("ThreadLeakageChecker exception:" + e.getStackTrace());
+    }
+    // Assert here does not work.
+    if (!status) {
+      System.out.println("---------- Test Class " + testClassName + " thread leakage detected! ---------------");
```

Comment:
Can we kill the threads and print log? At least, we keep the test working.

Reply:
This is going to keep the test continue to execute. Killing thread is a little bit dangerous as out detection logic may not 100% right.